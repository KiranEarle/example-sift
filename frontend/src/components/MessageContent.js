import React from 'react'
import PropTypes from 'prop-types'
import { getReturnPathDomain } from '../helpers/getReturnPathDomain'

const MessageContent = ({header, message, closeModal, fullDetails}) => {
  const sender = message.sender ? message.sender.email : 'No email sender details found'
  const spf = message.emailHeader ? message.emailHeader.spf : 'No SPF details found'
  const dkim = message.emailHeader ? message.emailHeader.dkim : 'No dkim details found'

  const returnPathDomain = getReturnPathDomain(dkim)

  return (
    <div>
      <div className='expanded-header'>
      < h2 className='message-header'>{header}</h2>
      </div>
      <p className='message-subject'>Subject: {message.subject}</p>
      {fullDetails &&
      <div className='full-details'>
      <ul>
        <li><span className='detail-title'>Word count:</span> {message.wordCount}</li>
        <li><span className='detail-title'>id:</span>  {message.id}</li>
        <li><span className='detail-title'>thread id:</span>  {message.threadId}</li>
      </ul>
      <h3 className='dmarc-title'>DMARC information</h3>
      <ul>
        <li><span className='detail-title'>from: </span>{sender}</li>
        <li><span className='detail-title'>Return Path Domain: </span>{returnPathDomain}</li>
        <li><span className='detail-title'>SPF: </span>{spf}</li>
        <li><span className='detail-title'>DKIM: </span>{dkim}</li>
      </ul>
      {closeModal &&
        <button onClick={() => closeModal()}>Close</button>
      }
      </div>}
    </div>
  )
}

MessageContent.PropTypes = {
  /**
   * takes a string for the header.
   */
  header: PropTypes.string,

  /**
   * takes an object with details of message.
   */
   message: PropTypes.object,

  /**
   * function that is passed in if the content is a modal
   * to be passed in as an onclick function on a button,
   * will not render the button if not passed as a prop.
   */
  closeModal: PropTypes.func,

  /**
   * boolean which checkes if user wants all the details
   * to be shown.
   */

}

export default MessageContent