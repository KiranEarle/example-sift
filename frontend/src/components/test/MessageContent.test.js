import React from 'react'
import MessageContent from '../MessageContent'

it('render with props without blowing up', () => {
  const header = 'text'
  const message = {
    body: 'text',
    id: "text",
    subject: "more text",
    threadId: "even more text",
    wordCount: 198
  }

  const component = shallow(<MessageContent header={header} message={message} />);
  expect(component).toMatchSnapshot();
});