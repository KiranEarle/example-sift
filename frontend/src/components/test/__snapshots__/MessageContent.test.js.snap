// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render with props without blowing up 1`] = `
<div>
  <div
    className="expanded-header"
  >
    <h2
      className="message-header"
    >
      text
    </h2>
  </div>
  <p
    className="message-subject"
  >
    Subject: 
    more text
  </p>
</div>
`;
